internal/service/backupcompliancepolicy/resource_backup_compliance_policy.go

```diff
@@ -8,13 +8,15 @@ import (
 	"net/http"
```

Comment:
Regarding the possibility of exposing `overwrite_backup_policies` attribute we can wait for confirmation from the API team but it would seem reasonable that only the UPDATE request receives this parameter.
With that said we can expose the attribute with the default value of false, and this will ensure that if a customer was relying on the existing behaviour they can quickly adjust to that behaviour with the explicit `overwrite_backup_policies = true`.

Reply:
Added [comment to ticket](https://jira.mongodb.org/browse/CLOUDP-229775?focusedCommentId=6198851&focusedId=6198851&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-6198851) about Backup team's response, this attribute is not persisted and hence cannot be returned in API response.
Hence the idea is to not expose this to user to be able to update

Not sure if this addresses your concern?

Reply:
My question is more related to why we would not add this attribute, because from technical stand point it does seem possible. We can define it as an optional attribute with a default value, include the value for patch requests and in the case of a read simply preserve the value defined in the tf config.
Personally I see value in exposing the attribute in case customers have preference for the existing behaviour of having their backup definitions automatically change to the new policy.

Reply:
@AgustinBettati  To make sure I understand correctly, you mean even though this attribute isn't returned from the API, we should persist in the TF state and re-use that every time for PATCH, is that correct? 
If yes, personally unless there's a strong reason, I think we should avoid having such attributes as these can be problematic during imports (for example if user does a `terraform state rm` and then wants to re-import for some reason or to get out of a consistent state)

>Personally I see value in exposing the attribute in case customers have preference for the existing behaviour of having their backup definitions automatically change to the new policy.

I understand this but I'm also thinking that:
1. this can potentially leave them in an inconsistent state (even if they intended the behavior)
2. and in addition of the inconsistent state, they now also have an additional attribute to deal with in case they want to re-import the resource for any reason.

I think we should only add this as a new attribute to be used by customers if we get a customer request for such a use-case. WDYT?

Reply:
`overwriteBackupPolicies` parameter was purely created to help downstream tools like AKO and TF to behave as intended. upstream Atlas API team did not intend for API users to manipulate directly as is evidenced that it is not return back in GET/UPDATE responses. in this case suggest we keep design simple and do not allow users to directly modify parameter. 

if this feedback comes in from a customer we can always re-evaluate, but for now prefer not to over engineer/complicate this resource and keep behavior consistent with other downstream tools that support BCP.

Reply:
Looping in @colm-quinn  for awareness since he was interested in this and had an opinion